src/Atc.Kepware.Configuration/Services/Connectivity/KepwareConfigurationClientConnectivity.cs

```diff
@@ -472,6 +502,95 @@ public sealed partial class KepwareConfigurationClient
             cancellationToken);
     }
 
+    private static HttpClientRequestResult<IList<TDevice>?> ProcessGetDevicesByChannelNameResponse<TDevice>(HttpClientRequestResult<IList<JsonObject>?> response)
+        where TDevice : DeviceBase
+    {
+        // No data, return early, nothing to adapt
+        if (!response.HasData)
+        {
+            return new HttpClientRequestResult<IList<TDevice>?>()
+            {
+                CommunicationSucceeded = response.CommunicationSucceeded,
+                StatusCode = response.StatusCode,
+                Message = response.Message,
+                Exception = response.Exception,
+            };
+        }
+
+        // No actual devices, return early, nothing to adapt
+        if (response.Data!.Count == 0)
+        {
+            return new HttpClientRequestResult<IList<TDevice>?>([])
+            {
+                CommunicationSucceeded = response.CommunicationSucceeded,
+                StatusCode = response.StatusCode,
+                Message = response.Message,
+                Exception = response.Exception,
+            };
+        }
+
+        if (!TryGetDeviceTypeJsonMappingType<TDevice>(out Type? jsonMappingType))
+        {
+            return new HttpClientRequestResult<IList<TDevice>?>()
+            {
+                CommunicationSucceeded = response.CommunicationSucceeded,
+                StatusCode = response.StatusCode,
+                Message = response.Message,
+                Exception = new NotSupportedException($"Could not find a JSON mapping type for {typeof(TDevice).Name}"),
+            };
+        }
+
+        // Serialize to JSON mapping type
+        IList<object?> mappedDeviceType = response
+            .Data
+            .Select(x => JsonSerializer.Deserialize(x.ToString(), jsonMappingType))
```

Comment:
Change to utilize jsonSerializerOptions:

.Select(x => JsonSerializer.Deserialize(x.ToString(), jsonMappingType, jsonSerializerOptions))

And non-static method.